core/templates/dev/head/pages/exploration_player/ImagePreloaderService.js

```diff
@@ -165,6 +204,14 @@ oppia.factory('ImagePreloaderService', [
         });
         if (noOfImagesNeitherInCacheNorDownloading > 0 &&
             noOfImageFilesCurrentlyDownloading <= 1) {
+          var imageFilesInGivenState = ExtractImageFilenamesFromStateService
+            .getImageFilenamesInState(_states.getState(stateName));
+          _filenamesOfImageFailedToDownload = _filenamesOfImageFailedToDownload
+            .map(function(filename) {
+              if (imageFilesInGivenState.indexOf(filename) === -1) {
+                return filename;
+              }
+            });
```

Comment:
Maybe use filter here.

Reply:
Shouldn't we move this into `_kickOffImagePreloader`?

Reply:
Done!